packages/@aws-cdk/aws-ecs/lib/container-image.ts

```diff
@@ -11,7 +18,7 @@ export abstract class ContainerImage {
    * Reference an image on DockerHub or another online registry
    */
   public static fromRegistry(name: string, props: RepositoryImageProps = {}) {
-    return new RepositoryImage(name, props);
+    return new RepositoryImage(name, { ...props, ecrRepository: ECR_IMAGE_REGEX.test(name) });
```

Comment:
Check that “name” is not a token (Token.isUnresolved) and only then match against the regex. Also move this test to bind()

Reply:
I think in most cases this will be a Token, or at least *contain* Tokens. Be prepared to deal with that.

Reply:
No problem, will address that. But could you explain why a token can also possibly match the regex?